alpha/apps/kaltura/lib/myContentStorage.class.php

```diff
@@ -69,6 +69,48 @@ public static function getGeneralEntityPath($entityName, $int_id, $id, $fileName
 		return $res;
 		
 	}
+
+	public static function getThumbEntityPath($entityName, $entry, $thumbName, $fileName, $version)
+	{
+		$currentDcId = kDataCenterMgr::getCurrentDcId();
+		if(myCloudUtils::isCloudDc($currentDcId))
+		{
+			$periodicStorageIds = kStorageExporter::getPeriodicStorageIds();
+			if($periodicStorageIds)
+			{
+				foreach($periodicStorageIds as $periodicStorageId)
+				{
+					$storage = StorageProfilePeer::retrieveByPK($periodicStorageId);
+					if($storage && $storage->getStorageBaseDir())
+					{
+						return $storage->getStorageBaseDir() . self::getThumbEntitySharedPath($entityName, $entry->getId(), $thumbName, $fileName, $version);
+					}
+				}
+			}
+		}
+
+		return self::getFSContentRootPath() . self::getGeneralEntityPath($entityName, $entry->getIntId(), $thumbName, $fileName, $version);
+	}
+
+	protected static function getThumbEntitySharedPath($entityName, $entryId, $thumbName, $fileName , $version = null)
+	{
+		if( $version != null )
```

Comment:
In case of version, would be easier to create the full new filename starting with version